### PR TITLE
feat(tabs): migrate tabs components

### PR DESCRIPTION
**Pull request by alexandereneroth**
**Describe pull-request**  
Migrates all the tabs components, improves upon stories, and extracts tab components color values to css variables.

**Solving issue**  
_In case of GitHub issue add # plus the number of the issue (for example #123) OR if it is Azure then AB# and number of ticket_
Fixes: [AB#2406](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/2406)

## `CHECKLIST`

Basic
- [x] Move component to tegel folder
- [x] Make it run
- [x] Make it look & work like old component
- [x] Clean up the imports 

Styles
- [x] Separate theming variables in component-theme-vars.scss 
- [x] Make sure to use variables instead of mixins for font + letter spacing 

Story
- [x] Change stories files from JS to TS 
- [x] Formater for code (gaps after condiotnal rendering ….) 
- [x] Number of stories for one component depends on complexity of component. Let Usage and Style be your guide for separation of complex component stories 
- [x] Link README for webComponents to story file (Check badges for guidance) 
- [x] Write name, description and defaultValue(optional) for each argument of story 
- [x] Think of conditional rendering of HTML for components segments (icons, etc) - make sure that user can easily copy/paste code from "Docs" tab 
- [x] Check if props or anything is missing 
- [x] Is the control type correct? Maybe the prop is a string but could be a boolean in storybook (ex showing a link in banner). More than 4 choices should be a dropdown instead of radio button. 
- [x] Make sure that there are default options so that a radio button is checked etc 
- [x] Name of controls and options. 
- [x] Maybe not all controls need to be visible in every story or at least be toggleable (boolean for advanced settings for example). Some control lists are really long.

### CODE REVIEW

File: tegel/src/components/tabs/inline-tabs/inline-tabs.stories.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  // TODO: change background color of stories globally with dark theme plugin instead
+  return formatHtmlPreview(`
+  <style>
+    /* .demo-wrapper is only added here for demonstration purposes. */
+    .demo-wrapper {
+      padding: 30px 20px;
+      font-size: 14px;
+      background-color: ${altBgColor ? getComputedStyle(document.documentElement).getPropertyValue('--sdds-grey-50') : 'white'};
+    }
+    .sdds-theme-dark .demo-wrapper {
```
Comment (mJarsater) [inline tegel/src/components/tabs/inline-tabs/inline-tabs.stories.ts:41]: Is this one still needed?

File: tegel/src/components/tabs/navigation-tabs/navigation-tabs.stories.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      <a href="#" class="sdds-navigation-tabs--tab__active">Active tab</a>
+      <a href="#">Tab name</a>
+      <a href="#">Tab name</a>
+      <a href="#">Tab name</a>
+      <a href="#" class="sdds-navigation-tabs--tab__disabled">Disabled tab</a>
+    </sdds-navigation-tabs>
+    `);
+
+export const Basic = Template.bind({});
+Basic.args = {};
```
Comment (mJarsater) [inline tegel/src/components/tabs/navigation-tabs/navigation-tabs.stories.ts:23]: Rename to Default.

File: tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    }
+  }
+
+  ::slotted(&--tab__active:focus)::after {
+    width: 100%;
+  }
+
+  ::slotted(&--tab__disabled) {
+    color: rgb(0 21 51 / 38%);
+    pointer-events: none;
```
Comment (mJarsater) [inline tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss:176]: Should we maybe add a cursor: not-allowed here?

Comment (alexandereneroth) [reply tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss:176]: @mJarsater Do we have this in other places?
@theJohnnyMe What do you think?

Comment (mJarsater) [reply tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss:176]: For the buttons we use it 😄

Comment (alexandereneroth) [reply tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss:176]: I suggest we make a separate issue for this to make it consistent everywhere.
I suspect there are other inconsistencies as well

Comment (mJarsater) [reply tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss:176]: Agree 👍🏻

Comment (mJarsater) [reply tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss:176]: Created a ticket here: https://dev.azure.com/scaniadigitaldesignsystem/SDDS/_backlogs/backlog/Product%20Team/Features/?workitem=2476

Comment (theJohnnyMe) [reply tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss:176]: As you talked here and yesterday during the grooming, let do it in separate ticket ;)

File: tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+      width: 100%;
+      background-color: var(--sdds-inline-tabs-fullbleed-tab-active-indicator-bg);
+    }
+  }
+
+  ::slotted(&--tab__active:focus)::after {
+    width: 100%;
+  }
+
+  ::slotted(&--tab__disabled) {
```
Comment (theJohnnyMe) [inline tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss:174]: Not sure if is it the right place, but please check if we can disable the focus state for disabled items.
Only this variation of tabs gets focus state on disabled item: 

Screenshot 2022-11-02 at 10 21 51

File: tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss [hunk 5]
```diff
@@ -0,0 +1,179 @@
+// TODO question: should we not share this like before?
+@mixin sdds-focus-state {
```
Comment (theJohnnyMe) [inline tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss:2]: Try to import this one from the mixins folder instead.

File: tegel/src/components/tabs/inline-tabs/inline-tabs.scss [hunk 6]
```diff
@@ -0,0 +1,196 @@
+// TODO question: should we not share this like before?
+@mixin sdds-focus-state {
```
Comment (theJohnnyMe) [inline tegel/src/components/tabs/inline-tabs/inline-tabs.scss:2]: Try to import this one from the mixins folder.

File: tegel/src/components/tabs/inline-tabs/inline-tabs.tsx [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+  @State() buttonWidth: number = 0;
+
+  /** current calculated tab height (calculated from the one with the most height) */
+  @State() tabHeight: number = 0;
+
+  @State() showLeftScroll: boolean = false;
+
+  @State() showRightScroll: boolean = false;
+
+  // public method for switching to a tab programmatically
```
Comment (theJohnnyMe) [inline tegel/src/components/tabs/inline-tabs/inline-tabs.tsx:33]: Please add comments here in a format as it is done for Prop above, with /** */ so it displays in README

File: tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.stories.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import { formatHtmlPreview } from '../../../utils/utils';
+import readme from './readme.md';
+
+export default {
+  title: 'Components/Tabs/Inline Tabs (full bleed)',
+  parameters: {
```
Comment (theJohnnyMe) [inline tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.stories.ts:6]: Maybe add a layout parameter for a fullscreen view, what do you think?

File: tegel/src/components/tabs/navigation-tabs/navigation-tabs.stories.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import { formatHtmlPreview } from '../../../utils/utils';
+import readme from './readme.md';
+
+export default {
+  title: 'Components/Tabs/Navigation Tabs',
+  parameters: {
+    notes: readme,
```
Comment (theJohnnyMe) [inline tegel/src/components/tabs/navigation-tabs/navigation-tabs.stories.ts:7]: Maybe also a fullscreen layout here, what do you think?

**Review by theJohnnyMe — CHANGES_REQUESTED** [review]

File: tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss (hunk 4 above)
Comment (alexandereneroth) [reply tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss:174]: I think we should rather make the other disabled controls focusable. If they are not, someone using a screenreader might not discover that they exist.
Saw this https://ux.stackexchange.com/a/103246

What do you think?

File: tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss (hunk 5 above)
Comment (alexandereneroth) [reply tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss]: fixed

File: tegel/src/components/tabs/inline-tabs/inline-tabs.scss (hunk 6 above)
Comment (alexandereneroth) [reply tegel/src/components/tabs/inline-tabs/inline-tabs.scss]: fixed

File: tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.stories.ts (hunk 8 above)
Comment (alexandereneroth) [reply tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.stories.ts:6]: Personally i like that there is a bit of padding around it in the example, i think you can see the edges of it better, also it looks more consistent with other stories 🤔

File: tegel/src/components/tabs/navigation-tabs/navigation-tabs.stories.ts (hunk 9 above)
Comment (alexandereneroth) [reply tegel/src/components/tabs/navigation-tabs/navigation-tabs.stories.ts:7]: Personally i like that there is a bit of padding around it in the example, i think you can see the edges of it better, also it looks more consistent with other stories 🤔

File: tegel/src/components/tabs/inline-tabs/inline-tabs.tsx (hunk 7 above)
Comment (alexandereneroth) [reply tegel/src/components/tabs/inline-tabs/inline-tabs.tsx]: fixed

File: tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss (hunk 4 above)
Comment (theJohnnyMe) [reply tegel/src/components/tabs/inline-tabs-fullbleed/inline-tabs-fullbleed.scss:170]: I did not know about this, but good that you came back with a comment. Then let's keep it.

**Review by theJohnnyMe — APPROVED** [review]
Nice work man, LGTM! ;)

**Pull request merged**